hosted/preparation/install_test.go

```diff
@@ -0,0 +1,247 @@
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	"github.com/rancher-sandbox/ele-testhelpers/rancher"
+	"github.com/rancher-sandbox/ele-testhelpers/tools"
+)
+
+var _ = Describe("Provision k3s cluster with Rancher", Label("install"), func() {
+	// Create kubectl context
+	// Default timeout is too small, so New() cannot be used
+	k := &kubectl.Kubectl{
+		Namespace:    "",
+		PollTimeout:  tools.SetTimeout(300 * time.Second),
+		PollInterval: 500 * time.Millisecond,
+	}
+
+	It("Install upstream k3s cluster", func() {
+		if proxy != "" {
+			By("Run local squid proxy in docker", func() {
+				// Configure proxy before K3s installation if requested
+				GinkgoLogr.Info("Starting squid proxy")
+
+				cwd, _ := os.Getwd()
+				GinkgoLogr.Info("Current working directory: " + cwd)
+
+				out, err := exec.Command("docker", "run", "-d", "--rm", "--name", "squid_proxy",
+					"--volume", cwd+"/squid.conf:/etc/squid/squid.conf",
+					"-p", "3128:3128", "ubuntu/squid").CombinedOutput()
+				GinkgoWriter.Println(string(out))
+				Expect(err).To(Not(HaveOccurred()))
+			})
+
+			By("Configure proxy in /etc/default/k3s", func() {
+				k3sConfigPath := "/etc/default/k3s"
+				k3sConfig := fmt.Sprintf(`HTTP_PROXY=http://%s
+HTTPS_PROXY=http://%s
+NO_PROXY=127.0.0.0/8,10.0.0.0/8,cattle-system.svc,172.16.0.0/12,192.168.0.0/16,.svc,.cluster.local`, proxyHost, proxyHost)
+				// Write the proxy configuration to the file as root
+				out, err := exec.Command("sh", "-c", fmt.Sprintf("echo '%s' | sudo tee %s", k3sConfig, k3sConfigPath)).CombinedOutput()
+				GinkgoWriter.Println(string(out))
+				Expect(err).To(Not(HaveOccurred()))
+			})
+		}
+
+		By("Installing K3s", func() {
+			// Get K3s installation script
+			fileName := "k3s-install.sh"
+			Eventually(func() error {
+				return tools.GetFileFromURL("https://get.k3s.io", fileName, true)
+			}, tools.SetTimeout(2*time.Minute), 10*time.Second).ShouldNot(HaveOccurred())
+
+			// Set command and arguments
+			installCmd := exec.Command("sh", fileName)
+			// INSTALL_K3S_VERSION should be already set by the action or user's shell
+			installCmd.Env = append(os.Environ(), "INSTALL_K3S_EXEC=--disable metrics-server --write-kubeconfig-mode 644")
+
+			// Retry in case of (sporadic) failure...
+			count := 1
+			Eventually(func() error {
+				// Execute K3s installation
+				out, err := installCmd.CombinedOutput()
+				GinkgoWriter.Printf("K3s installation loop %d:\n%s\n", count, out)
+				count++
+				return err
+			}, tools.SetTimeout(2*time.Minute), 5*time.Second).Should(BeNil())
+		})
+
+		By("Starting K3s", func() {
+			err := exec.Command("sudo", "systemctl", "start", "k3s").Run()
+			Expect(err).To(Not(HaveOccurred()))
+
+			// Delay few seconds before checking
+			time.Sleep(tools.SetTimeout(5 * time.Second))
+		})
+
+		By("Waiting for K3s to be started", func() {
+			// Wait for all pods to be started
+			checkList := [][]string{
```

Comment:
I think somewhere around here you need to ensure a KUBECONFIG env var is exported. When I tried running this locally without a KUBECONFIG exported, it errored out.

Perhaps, you can add KUBECONFIG to the `REQUIRED_VARS` list and also ensure it is `.ToNot(BeEmpty())` in the `BeforeSuite`?